modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/spi/model/index/contributor/test/ExternalReferenceCodeModelDocumentContributorTest.java

```diff
@@ -0,0 +1,176 @@
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.search.engine.adapter.SearchEngineAdapter;
+import com.liferay.portal.search.engine.adapter.search.CountSearchRequest;
+import com.liferay.portal.search.engine.adapter.search.CountSearchResponse;
+import com.liferay.portal.search.query.BooleanQuery;
+import com.liferay.portal.search.query.Queries;
+import com.liferay.portal.search.query.TermQuery;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import java.util.Objects;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Joshua Cords
+ */
+@RunWith(Arquillian.class)
+public class ExternalReferenceCodeModelDocumentContributorTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			TestPropsValues.getGroupId(), TestPropsValues.getUserId());
+
+		_blogsEntry = BlogsTestUtil.addEntryWithWorkflow(
+			TestPropsValues.getUserId(), RandomTestUtil.randomString(), false,
+			_serviceContext);
+
+		_journalFolder = JournalFolderServiceUtil.addFolder(
+			null, TestPropsValues.getGroupId(), 0,
+			RandomTestUtil.randomString(), StringPool.BLANK, _serviceContext);
+
+		_journalArticle = JournalTestUtil.addArticle(
+			TestPropsValues.getGroupId(), 0,
+			PortalUtil.getClassNameId(JournalArticle.class),
+			HashMapBuilder.put(
+				LocaleUtil.US, RandomTestUtil.randomString()
+			).build(),
+			null,
+			HashMapBuilder.put(
+				LocaleUtil.US, ""
+			).build(),
+			LocaleUtil.getSiteDefault(), false, true, _serviceContext);
+
+		_user = UserTestUtil.addUser(TestPropsValues.getGroupId());
+	}
+
+	@Test
+	public void testExternalReferenceCodeContributed() throws Exception {
+		_assertERCSearch(_blogsEntry);
+		_assertERCSearch(_journalArticle);
+		_assertERCSearch(_journalFolder);
+		_assertERCSearch(_user);
+	}
+
+	private void _assertERCSearch(
+			ExternalReferenceCodeModel externalReferenceCodeModel)
+		throws Exception {
+
+		TermQuery companyTermQuery = _queries.term(
+			Field.COMPANY_ID, TestPropsValues.getCompanyId());
+
+		TermQuery externalReferenceCodeQuery = _queries.term(
+			"externalReferenceCode",
+			externalReferenceCodeModel.getExternalReferenceCode());
+
+		BooleanQuery booleanQuery = _queries.booleanQuery();
+
+		booleanQuery.addMustQueryClauses(
+			companyTermQuery, externalReferenceCodeQuery);
```

Comment:
inline both variables companyTermQuery and externalReferenceCodeQuery as they are used once.